Index: mt7603e-p4rev-112670/src/os/linux/rt_linux.c
===================================================================
--- mt7603e-p4rev-112670.orig/src/os/linux/rt_linux.c	2016-05-10 22:14:55.546377000 +0800
+++ mt7603e-p4rev-112670/src/os/linux/rt_linux.c	2016-05-10 22:15:42.598375021 +0800
@@ -322,7 +322,7 @@
 #ifdef RA_MTD_RW_BY_NUM
 	ra_mtd_read(MTD_NUM_FACTORY, 0, (size_t) b, p);
 #else
-	ra_mtd_read_nm("Factory", a&0xFFFF, (size_t) b, p);
+	ra_mtd_read_nm("factory", a&0xFFFF, (size_t) b, p);
 #endif
 #endif /* CONFIG_RALINK_FLASH_API */
 }
@@ -338,7 +338,7 @@
 #ifdef RA_MTD_RW_BY_NUM
 	ra_mtd_write(MTD_NUM_FACTORY, 0, (size_t) b, p);
 #else
-	ra_mtd_write_nm("Factory", a&0xFFFF, (size_t) b, p);
+	ra_mtd_write_nm("factory", a&0xFFFF, (size_t) b, p);
 #endif
 #endif /* CONFIG_RALINK_FLASH_API */
 }
